app/models/miq_expression/expression/base.rb

```diff
@@ -0,0 +1,3 @@
+class MiqExpression::Expression::Base
+  delegate :to_sql, :to => :to_arel
```

Comment:
If we choose to make objects of these things, we should really add abstract methods for all things the caller expects.

I believe they all need to implement build and to_arel.